core/thread_pool_test.cpp

```diff
@@ -57,6 +57,13 @@ TEST(ThreadPool, SimpleTaskWithArgs)
     EXPECT_EQ(second, sum);
 }
 
+// We need to capture first because otherwise MSVC complains about
+// an implicit capture. However, this makes clang sad, so we have to
+// ignore its warning.
+#if defined(__clang__)
+#pragma clang diagnostic ignored "-Wunused-lambda-capture"
```

Comment:
Why not doing that at the CMake level?

Reply:
I wasn't sure whether to apply this everywhere. I was hoping it only ever comes up here.

Reply:
As you want :).

As a principle, I don't like macros in the code, and I prefer to keep my code independent from the build system (as much as possible). But I'm fine with this as well.

Reply:
Convinced :smile: